### ISSUE TITLE
How do I get access to private repos (Gitlab) 

### ISSUE THREAD

**Issue by weeebdev** [git-touch/git-touch#225]
The app doesn't read private repos that I don't own, but has been added as Developer or Maintainer.

**Comment by timo955** [comment]
https://github.com/google/boringssl/blob/238a25831e598642befcf056e778c8f166781a97/util/BUILD.toplevel

**Comment by timo955** [comment]
_Sent from my LM-Q730 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by weeebdev** [comment]
> 
> 
> _Sent from my LM-Q730 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

???

**Comment by Lukas-Batema** [comment]
> The app doesn't read private repos that I don't own, but has been added as Developer or Maintainer. 

I will check to see if I get the same results. What version are you running, what device, etc.

**Comment by weeebdev** [comment]
> > The app doesn't read private repos that I don't own, but has been added as Developer or Maintainer. 
> 
> I will check to see if I get the same results. What version are you running, what device, etc.
> ___
> EDIT: 
> ___
> Go to: `Settings/Review Permissions`
> ___
> Check near the bottom of the page for a line that says something like, `Full control of private repositories`. If you do not see that, please redo your permissions to allow this. Hope that helped. If not, please let me know!
> ___
> You might have logged in originally using the option with `OAuth (Public Repos Only)` If you remember doing so, you may need to remove your account from the list, temporarily revoke access in the settings directory I mentioned earlier, then log in with the option using just `OAuth` (without any special features). 
> ___
> The first option seems more probable, according to your explanation of the issue.
> ___

I am sorry, but I think I didn't understand you. I meant Gitlab, so I can't find `Review Permissions ` there. Also, for gitlab it's only possible to login via tokens, not OAuth and I already grant all necessary permissions like api, read_user and read_repository

**Comment by Lukas-Batema** [comment]
> > > The app doesn't read private repos that I don't own, but has been added as Developer or Maintainer. 
> > 
> > I will check to see if I get the same results. What version are you running, what device, etc.
> > ___
> > EDIT: 
> > ___
> > Go to: `Settings/Review Permissions`
> > ___
> > Check near the bottom of the page for a line that says something like, `Full control of private repositories`. If you do not see that, please redo your permissions to allow this. Hope that helped. If not, please let me know!
> > ___
> > You might have logged in originally using the option with `OAuth (Public Repos Only)` If you remember doing so, you may need to remove your account from the list, temporarily revoke access in the settings directory I mentioned earlier, then log in with the option using just `OAuth` (without any special features). 
> > ___
> > The first option seems more probable, according to your explanation of the issue.
> > ___
> 
> I am sorry, but I think I didn't understand you. I meant Gitlab, so I can't find `Review Permissions ` there. Also, for gitlab it's only possible to login via tokens, not OAuth and I already grant all necessary permissions like api, read_user and read_repository

My bad, sorry.

**Comment by weeebdev** [comment]
> > > > The app doesn't read private repos that I don't own, but has been added as Developer or Maintainer.
> > > 
> > > 
> > > I will check to see if I get the same results. What version are you running, what device, etc.
> > > 
> > > EDIT:
> > > 
> > > Go to: `Settings/Review Permissions`
> > > 
> > > Check near the bottom of the page for a line that says something like, `Full control of private repositories`. If you do not see that, please redo your permissions to allow this. Hope that helped. If not, please let me know!
> > > 
> > > You might have logged in originally using the option with `OAuth (Public Repos Only)` If you remember doing so, you may need to remove your account from the list, temporarily revoke access in the settings directory I mentioned earlier, then log in with the option using just `OAuth` (without any special features).
> > > 
> > > The first option seems more probable, according to your explanation of the issue.
> > 
> > 
> > I am sorry, but I think I didn't understand you. I meant Gitlab, so I can't find `Review Permissions ` there. Also, for gitlab it's only possible to login via tokens, not OAuth and I already grant all necessary permissions like api, read_user and read_repository
> 
> My bad, sorry.

No, it's my bad, sorry. I'll edit my issue to make it more obvious 😄

**Comment by shreyas1599** [comment]
@weeebdev can you verify with this endpoint if the repos you wish to be fetched are visible? I'm not a member of any gitlab project so I can't verify. https://gitlab.com/api/v4/users/{your_user_id}/projects?membership=true&Private-Token={your_private_token}. If you aren't aware of your user id you can get it from https://gitlab.com/api/v4/users?username={your_gitlab_username}&Private-Token={your_private_token}
Thanks.

**Comment by weeebdev** [comment]
> [https://gitlab.com/api/v4/users/{your_user_id}/projects?membership=true&Private-Token={your_private_token}](https://gitlab.com/api/v4/users/%7Byour_user_id%7D/projects?membership=true&Private-Token=%7Byour_private_token%7D)

Nope. I do only see my own public repos (not even private)
